### ISSUE TITLE
The slider of the query result view is useless in case of long tables

### ISSUE THREAD

**Issue by akshay-joshi** [akshay-joshi/TestRMToHub#228]
**This issue has been migrated from Redmine.**

Author Name: **bassi bass** (bassi bass)
Original Redmine Issue: [3269](https://redmine.postgresql.org/issues/3269)

---
If I use the Slider on the right side of the result vies they would jump up if new lines are loaded. Then I have grab the slider again to navigate in the result set. This is very irritating if I navigate within long results. Going to the end of a long result with this way is nearly impossible.

---

- [Screenshot 2020-04-22 at 10.50.01 AM.png](https://redmine.postgresql.org/attachments/download/4845/Screenshot%202020-04-22%20at%2010.50.01%20AM.png) (Pradip Parkale)
- [Screenshot 2020-04-23 at 11.27.24 AM.png](https://redmine.postgresql.org/attachments/download/4853/Screenshot%202020-04-23%20at%2011.27.24%20AM.png) (Shubham Agarwal)

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
For a future release, *if* we can get the final row count without loading all the rows, then we could potentially adjust the scrollbar to reflect the correct number of rows from the outset. However, if we can't do this without loading all rows, then we'll leave it as-is.

However, a workaround if desired is to create a conflg_local.py file alongside pgAdmin's config.py file, and in there add a line such as:

ON_DEMAND_RECORD_COUNT = 10000000

This will make the batch load size 10M rows which in most sane cases will cause all rows to be loaded at once, so the scrollbar will be sized correctly, though of course that may effectively slow down result retrieval. Obviously the value used here can be adjusted to suit preferences.

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)



---

-   Sprint  was changed from    to Ready

**Comment by akshay-joshi** [comment]
Author Name: **Aditya Toshniwal** (@adityatoshniwal)



---

-   Sprint  was changed from   Ready   to squash

**Comment by akshay-joshi** [comment]
Author Name: **Nagesh Dhope** (@dhope-nagesh)



---

-  assigned_to_id  was configured    as 7141 
-  status_id  was changed from   1   to 2

**Comment by akshay-joshi** [comment]
Author Name: **Nagesh Dhope** (@dhope-nagesh)


---
Applied in changeset 8d4f3c07c777d57cf0e89af8ed06b0e7886f37ab.

---

-  done_ratio  was changed from   0   to 90 
-  status_id  was changed from   2   to 7

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)



---

-  fixed_version_id  was configured    as 104

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)



---

-  assigned_to_id  removed   7141

**Comment by akshay-joshi** [comment]
Author Name: **Pradip Parkale** (@pradipparkale269)



---

-  assigned_to_id  was configured    as 8545

**Comment by akshay-joshi** [comment]
Author Name: **Pradip Parkale** (@pradipparkale269)


---
Hi Nagesh,

I have tested this on the latest snapshot build https://www.postgresql.org/ftp/pgadmin/pgadmin4/snapshots/2020-04-22/.

I created a table with 1000100 entries and hit select * from table then few rows get loaded and others were showing 'loading', I waited around an hour but no new row was loaded.

Please see the attached screenshot.


---

-  status_id  was changed from   7   to 1 
-  4845  was configured    as Screenshot 2020-04-22 at 10.50.01 AM.png 

---

- [Screenshot 2020-04-22 at 10.50.01 AM.png](https://redmine.postgresql.org/attachments/download/4845/Screenshot%202020-04-22%20at%2010.50.01%20AM.png) (Pradip Parkale)

**Comment by akshay-joshi** [comment]
Author Name: **Pradip Parkale** (@pradipparkale269)



---

-  assigned_to_id  removed   8545

**Comment by akshay-joshi** [comment]
Author Name: **Nagesh Dhope** (@dhope-nagesh)



---

-  assigned_to_id  was configured    as 7141 
-  status_id  was changed from   1   to 2

**Comment by akshay-joshi** [comment]
Author Name: **Nagesh Dhope** (@dhope-nagesh)


---
Applied in changeset 5e91ed2bb1123b7bc7a96ee783d50b8b8b2f716b.

---

-  status_id  was changed from   2   to 7

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)



---

-  assigned_to_id  removed   7141

**Comment by akshay-joshi** [comment]
Author Name: **Shubham Agarwal** (@edbshubham)



---

-  assigned_to_id  was configured    as 4428

**Comment by akshay-joshi** [comment]
Author Name: **Shubham Agarwal** (@edbshubham)


---
The values are not getting loaded after 2000 rows. Attached is the screenshot.

---

-  assigned_to_id  was changed from   4428   to 7141 
-  status_id  was changed from   7   to 2 
-  4853  was configured    as Screenshot 2020-04-23 at 11.27.24 AM.png 

---

- [Screenshot 2020-04-23 at 11.27.24 AM.png](https://redmine.postgresql.org/attachments/download/4853/Screenshot%202020-04-23%20at%2011.27.24%20AM.png) (Shubham Agarwal)

**Comment by akshay-joshi** [comment]
Author Name: **Akshay Joshi** (@akshay-joshi)


---
Reverting patch for RM #3269.
    
We observed that sometimes the browser is getting hanged and sometimes the ViewData grid is getting disappear.
We suspect its due to the number of rows to update on the slick grid after fetching next of rows.

---

-   Sprint  was changed from   squash   to Ready 
-  done_ratio  was changed from   90   to 0 
-  fixed_version_id  removed   104  
-  assigned_to_id  removed   7141  
-  status_id  was changed from   2   to 1
